src/services/WalletService/deterministic/trezor.ts

```diff
@@ -31,11 +31,15 @@ export class TrezorWallet extends HardwareWallet {
     });
   }
 
-  public signRawTransaction(tx: EthTx): Promise<Buffer> {
+  public signRawTransaction(tx: UnsignedTransaction): Promise<Buffer> {
     return new Promise((resolve, reject) => {
-      const { chainId, ...strTx } = getTransactionFields(tx);
+      const { chainId, ...rest } = tx;
+      if (!chainId) {
+        return reject(Error('Missing chainId on tx'));
+      }
+      const formattedTx = { ...rest, nonce: hexlify(rest.nonce!) };
```

Comment:
Is the nonce checked before this function is called? Otherwise maybe check if a nonce is set.